### PR TITLE
Calling curl unpause from callback thread

### PR DESCRIPTION
**Pull request by MushMal**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/source/Response.c [hunk 1]
```diff
@@ -655,7 +654,11 @@ SIZE_T postReadCallback(PCHAR pBuffer, SIZE_T size, SIZE_T numItems, PVOID custo
     pCurlApiCallbacks = pCurlRequest->pCurlApiCallbacks;
     uploadHandle = pCurlResponse->pCurlRequest->uploadHandle;
 
-    if (pCurlResponse->paused) {
+    UINT64 curTime = GETTIME();
+    DLOGE("MMMMMMM postReadTime diff %llu", curTime - pCurlResponse->time);
```
[the remark this answers is not in the document]
Comment (MushMal) [reply src/source/Response.c:658]: Oh, this is a private PR - just need to see how this works and get some opinion from @chehefen . I wasn't sure how to mark it as WIP

File: src/source/Response.c [hunk 2]
```diff
@@ -752,3 +755,48 @@ SIZE_T postReadCallback(PCHAR pBuffer, SIZE_T size, SIZE_T numItems, PVOID custo
...
 }
+
+INT32 progressCallback(PVOID customData, curl_off_t dltotal, curl_off_t dlnow,
+        curl_off_t ultotal, curl_off_t ulnow)
+{
+    UNUSED_PARAM(dltotal);
+    UNUSED_PARAM(dlnow);
+    UNUSED_PARAM(ultotal);
+    UNUSED_PARAM(ulnow);
+    DLOGV("progressCallback (curl callback) invoked");
```
[the remark this answers is not in the document]
Comment (MushMal) [reply src/source/Response.c:766]: This was in the same style as other CURL callbacks. Will be re-doing perhaps

File: src/source/Response.c [hunk 3]
```diff
@@ -752,3 +755,48 @@ SIZE_T postReadCallback(PCHAR pBuffer, SIZE_T size, SIZE_T numItems, PVOID custo
...
+
+    // Check if we need to unpause the curl
+    expected = TRUE;
+    if (ATOMIC_COMPARE_EXCHANGE_BOOL(&pCurlResponse->unpause, &expected, (SIZE_T) FALSE)) {
+        // un-pause curl if paused
+        expected = TRUE;
+        if (ATOMIC_COMPARE_EXCHANGE_BOOL(&pCurlResponse->paused, &expected, (SIZE_T) FALSE)) {
+            result = curl_easy_pause(pCurlResponse->pCurl, CURLPAUSE_SEND_CONT);
+            if (result != CURLE_OK) {
+                DLOGW("Failed to un-pause curl with error: %u", result);
```
[the remark this answers is not in the document]
Comment (MushMal) [reply src/source/Response.c:793]: Good idea!

[the remark this answers is not in the document]
Comment (MushMal) [reply src/source/Response.c:793]: I am going to finalize on the entire approach and will make changes once done

File: src/source/Response.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (disa6302) [reply src/source/Response.c:658]: You can select convert to draft on the side to mark it WIP.

**Comment by daveisfera** [comment]
Would this resolve the curl warnings to print on initialize of the KVS Producer SDK? If so, could this be cleaned up and merged?

**Pull request closed without merging**